app/permissions/my_module_repository_row.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+Canaid::Permissions.register_for(MyModuleRepositoryRow) do
+  can :update_stock_my_module_repository_row_stock_consumption do |user, my_module_repository_row|
```

Comment:
Why do we need it?

Reply:
Because it's a combination of states/permissions of the module, the row, and the repository. Otherwise we'd need to add the archived conditions at every point where we check the update consumption permission.

Reply:
fixed